src/default-directory-searcher.coffee

```diff
@@ -0,0 +1,107 @@
+{EventEmitter} = require 'events'
+Task = require './task'
+
+# Maintain a queue of ids of searches to run. When a search is complete, the active
+# search is cleared and the next search (if any) in the queue is notified to run.
+# This ensures there is at most one scan-handler task running at a time.
+searchQueue = []
+nextId = 1
+activeSearchId = 0
+emitter = new EventEmitter
```

Comment:
I'd like to avoid using global state here. I'm open to any alternatives you can think of, but my preference would be to just let `DirectorySearcher::search` take multiple directories, so we don't need queuing.